src/aiida/tools/dumping/processes.py

```diff
@@ -0,0 +1,448 @@
+    """Dump inputs of a `CalcJobNode` of type `SinglefileData` and `FolderData`.
+
+    :param calcjob_node: The `CalcJobNode` whose inputs will be dumped.
+    :type calcjob_node: CalcJobNode
+    :param output_path: The path where the inputs will be dumped.
+    :type output_path: Path
+    """
+    dump_types = (SinglefileData, FolderData)
+
+    # ? Not using the `node_class` argument of `get_incoming`, as it does not actually retrieve, e.g. a `UpfData` node
+    # ? (due to planned deprecation?)
+    # ? Instead, check for isinstance of `SinglefileData`
+    input_node_triples = calcjob_node.base.links.get_incoming(link_type=LinkType.INPUT_CALC)
+
+    for input_node_triple in input_node_triples:
+        # Select only repositories that hold objects and are of the selected dump_types
+        if len(input_node_triple.node.base.repository.list_objects()) > 0 and isinstance(
+            input_node_triple.node, dump_types
+        ):
+            # input_node_path = output_path / Path('node_inputs') / Path(input_node_triple.link_label)
+            input_node_path = output_path / inputs_relpath / Path(*input_node_triple.link_label.split('__'))
+            input_node_triple.node.base.repository.copy_tree(input_node_path)
+
+
+def calcjob_presubmit_dump(calcjob_node: CalcJobNode, output_path: Path):
+    """
+    Dump inputs of a `CalcJobNode` using the `presubmit` function.
+
+    :param process: The `CalcJobNode` whose inputs need to be dumped.
+    :type process: CalcJobNode
+    :param output_path: The path where the inputs will be dumped.
+    :type output_path: Path
+    """
+
+    builder_restart = calcjob_node.get_builder_restart()
+    runner = get_manager().get_runner()
+    calcjob_process: CalcJob = instantiate_process(runner, builder_restart)  # type: ignore[assignment]
+
+    # `presubmit` calls `prepare_for_submission` internally
+    calc_info = calcjob_process.presubmit(folder=Folder(abspath=output_path))
+
+    try:
+        # Hackish way to modify local copy list so that the pseudos are actually dumped where I want them to. Otherwise
+        # they
+        # end up in home...
+        local_copy_list = calc_info['local_copy_list'].copy()
+        # print('LOCAL_COPY_LIST', local_copy_list)
+        new_local_copy_list = [tuple(list(local_copy_list[0][:2]) + [str(output_path / local_copy_list[0][-1])])]
+        calc_info['local_copy_list'] = new_local_copy_list
+        # print('NEW_LOCAL_COPY_LIST', new_local_copy_list)
+    except IndexError:
+        # This happens if `local_copy_list` is empty
+        pass
+
+    local_transport = LocalTransport().open()
+    new_calcjob_node: CalcJobNode = calcjob_process.node
+    upload_calculation(
+        node=new_calcjob_node,
+        transport=local_transport,
+        calc_info=calc_info,
+        folder=Folder(abspath=output_path),
+        inputs=calcjob_process.inputs,
+        dry_run=False,
+    )
```

Comment:
prefer to use the context manager that automatically manages the closign as well
```suggestion
    with LocalTransport() as transport:
        new_calcjob_node: CalcJobNode = calcjob_process.node
        upload_calculation(
            node=new_calcjob_node,
            transport=transport,
            calc_info=calc_info,
            folder=Folder(abspath=output_path),
            inputs=calcjob_process.inputs,
            dry_run=False,
        )
```

Reply:
👍